From 3042eb3ee2f1bd315f5db139a4ef9f5027208987 Mon Sep 17 00:00:00 2001
From: Victor Wan <victor.wan@amlogic.com>
Date: Thu, 20 Mar 2014 15:16:13 +0800
Subject: [PATCH 3792/5965] meson8: clock: choose another oscillator when
 syspll is not locked

We have two oscillators in System PLL, one for high frequency and one for low.
The boundary between two shifts a lot for different wafer lot. So in the code,
if system PLL can't lock in the boundary area, change to another oscillator.
---
 arch/arm/mach-meson8b/clock.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-meson8b/clock.c b/arch/arm/mach-meson8b/clock.c
index c7f45013c7eb..03a90fe391c4 100644
--- a/arch/arm/mach-meson8b/clock.c
+++ b/arch/arm/mach-meson8b/clock.c
@@ -1159,6 +1159,9 @@ SETPLL:
 					aml_set_reg32_bits(P_HHI_SYS_PLL_CNTL4, 0, 10, 1);
 					printk(KERN_ERR"  HHI_SYS_PLL_CNTL4: %08x\n", 
 						aml_read_reg32(P_HHI_SYS_PLL_CNTL4));
+				}else{
+					latency.b.afc_dsel_bp_in = !latency.b.afc_dsel_bp_in;
+					printk(KERN_ERR"  INV afc_dsel_bp_in, new latency=%08x\n",latency.d32);
 				}
 				printk(KERN_ERR"  Try again!\n");
 				goto SETPLL;
-- 
2.19.0

